src/pixelator/graph/backends/implementations.py

```diff
@@ -628,157 +697,335 @@ def add_names_to_vertexes(self, vs_names: List[str]) -> None:
 
     def subgraphs(self) -> Iterable[Graph]:
         """Get subgraphs of each cluster."""
         from pixelator.graph import Graph
 
-        return [Graph(IgraphGraphBackend(g)) for g in self._raw.subgraphs()]
+        return [
+            Graph(IgraphGraphBackend(g)) for g in self._vertex_clustering.subgraphs()
+        ]
+
+
+class NetworkxBasedVertex(Vertex):
+    """A Vertex instance that plays well with NetworkX."""
+
+    def __init__(self, index: int, data: Dict):
+        """Create a new NetworkxBasedVertex instance."""
+        self._index = index
+        self._data = data
+
+    @property
+    def index(self):
+        """Get the index of the vertex."""
+        return self._index
+
+    @property
+    def data(self) -> Dict:
+        """Get the of the vertex as a dict."""
+        return self._data
+
+    def __getitem__(self, attr: str) -> Any:
+        """Get the attr of the provided vertex."""
+        return self._data[attr]
+
+    def __setitem__(self, attr: str, value: Any) -> None:
+        """Set the attr of the vertex."""
+        self._data[attr] = value
+
+
+class NetworkxBasedEdge(Edge):
+    """An Edge instance backed by a Networkx Edge."""
+
+    def __init__(
+        self, edge_tuple: Tuple[int, int, Any], graph: Union[nx.Graph, nx.MultiGraph]
+    ):
+        """Create a NetworkxBasedEdge instance."""
+        self.edge_tuple = edge_tuple
+        self._graph = graph
+
+    def __eq__(self, other: object) -> bool:
+        """Determine equality of edge and `other`."""
+        if isinstance(other, NetworkxBasedEdge):
+            return self.index == other.index
+        return False
+
+    def __hash__(self) -> int:
+        """Compute the hash of the edge."""
+        return hash(self.index)
+
+    @property
+    def index(self) -> int:
+        """The index of the edge."""
+        return self.edge_tuple[2]["index"]
+
+    @property
+    def vertex_tuple(self) -> Tuple[Vertex, Vertex]:
+        """Return the vertices the edge connects as a tuple."""
+        v1_idx, v2_idx, _ = self.edge_tuple
+        node_1_data = self._graph.nodes[v1_idx]
+        node_2_data = self._graph.nodes[v2_idx]
+        return (
+            NetworkxBasedVertex(v1_idx, node_1_data),
+            NetworkxBasedVertex(v2_idx, node_2_data),
+        )
 
 
 class NetworkxBasedVertexSequence(VertexSequence):
     """Proxy for a networkx based vertex sequence."""
 
-    def __init__(self, raw) -> None:
+    def __init__(self, vertices: Iterable[NetworkxBasedVertex]) -> None:
         """Instantiate a new NetworkxBasedVertexSequence."""
-        self._raw = raw
+        self._vertices: Dict[int, NetworkxBasedVertex] = {v.index: v for v in vertices}
 
     def __len__(self) -> int:
         """Get the number of vertexes."""
-        return len(self._raw)
+        return len(self._vertices.keys())
 
-    def select(self, **kwargs):
-        """Select a subset of vertices."""
-        raise NotImplementedError()
+    def vertices(self):
```

Comment:
SHould we write the return of the method?

Reply:
Fixed!